lib/motions/search-motion.coffee

```diff
@@ -60,6 +62,9 @@ class SearchBase extends MotionWithInput
 
     modFlags = Object.keys(modifiers).join('')
 
+    # Escape the term for use in regex
+    term = term.replace /([$])/g, '\\$1'
+
```

Comment:
This would prevent the use of `$` to mean the end of the line in normal searches.
I would actually suggest to override `getSearchTerm` in `SearchCurrentWord` so that it simply returns `new RegExp(_.escapeRegExp(term), 'g')` - or with `'gi'` instead of `'g'` if case insensitivity is desired for `SearchCurrentWord` and its kin.